src/python/google_cloud_utils/storage.py

```diff
@@ -854,6 +854,17 @@ def list_blobs(cloud_storage_path, recursive=True):
     yield blob['name']
 
 
+@retry.wrap(
+    retries=DEFAULT_FAIL_RETRIES,
+    delay=DEFAULT_FAIL_WAIT,
+    function='google_cloud_utils.storage.list_blobs')
+def list_blobs_full(cloud_storage_path, recursive=True):
```

Comment:
I think it might be cleaner to get callers to do the concatenation to get the full bucket path rather than add another function with mostly duplicated code?

Maybe add something like storage.join_path(), which helps concatenate GCS paths, and can be used in more places?

Reply:
Done!